src/pyaro/timeseries/Filter.py

```diff
@@ -1381,7 +1381,7 @@ def filter_stations(self, stations: dict[str, Station]) -> dict[str, Station]:
             if self._update_topo_file_path(lat, lon) or topo is None:
                 if topo is not None:
                     topo.close()
-                topo = xr.open_dataset(self._topo_file)
+                topo = xr.load_dataset(self._topo_file)
```

Comment:
Great, this removes the need of `with` around the `open_dataset`. The code
```
                if topo is not None:
                    topo.close()
```
is no longer needed.

The `topo = None` / `or topo is None` clauses are just another variable for checking if if the self._topo_file variable is changed.  So, setting `self._topo_file = None` instead of `topo = None` avoids using two variables for the same thing and we don't need the `or`.

Reply:
With `topo = None` it fells a bit less like "action at a distance", but I've reset `_topo_file` before the loop now.

Reply:
I agree that the usage of `self._topo_file`  here and in `self._update_topo_file_path` should be implemented much cleaner (e.g. by moving `self._topo_file` from object scope to function scope and sending it as variable to `self._update_topo_file_path`), but I leave that work to @thorbjoernl .